### ISSUE TITLE
bcftools v1.12 norm mishandles NEGATIVE_TRAIN_SITE and POSITIVE_TRAIN_SITE

### ISSUE THREAD

**Issue by slzarate** [samtools/bcftools#1501]
First identified with this error:
```
> bcftools view -q 1 file_2.vcf > file_3.vcf
[W::vcf_parse] Contig '��;QD=30.18;ReadPosRankSum=-1.871;SOR=0.848;VQSLOD=-106;culprit=DP' is not defined in the header. (Quick workaround: index the file with tabix.)
[W::vcf_parse_filter] FILTER '1:17,17:94:99:2466,0' is not defined in the header
[E::bcf_hdr_parse_line] Could not parse the header line: "##FILTER=<ID=1:17,17:94:99:2466,0,Description=\"Dummy\">"
[E::vcf_parse_filter] Could not add dummy header for FILTER '1:17,17:94:99:2466,0' at ��;QD=30.18;ReadPosRankSum=-1.871;SOR=0.848;VQSLOD=-106;culprit=DP:0
[E::bcf_write] Broken VCF record, the number of columns at chrY:11533248 does not match the number of samples (0 vs 1286)
[main_vcfview] Error: cannot write to (null)
```

`file_2.vcf` was generated using the command `bcftools norm -a --atom-overlaps . -m -both file_1.vcf > file_2.vcf`. The lines around the offending lines are:

```
> grep -A 2 -B 2 'm;QD=30.18;ReadPosRankSum=-1.871;SOR=0.848;VQSLOD=-106;culprit=DP' file_2.vcf
chrY    11516414        .       G       A       931251  PASS    AC=1285;AF=1;AN=1285;BaseQRankSum=4.8;DP=27120;FS=0;MLEAC=1599;MLEAF=1;MQ=59.99;MQ0=0;MQRankSum=0;QD=28.86;ReadPosRankSum=-0.108;SOR=0.773;VQSLOD=0.959;culprit=DP
chrY    11533248        .       C       T       4.64891e+06     PASS    AC=1285;AF=1;AN=1285;BaseQRankSum=0.353;DP=161647;FS=1.579;MLEAC=1599;MLEAF=1;MQ=40.52;MQ0=0;MQRankSum=-5.845;NEGATIVE_TRAIN_SITE==
��&�:m;QD=30.18;ReadPosRankSum=-1.871;SOR=0.848;VQSLOD=-106;culprit=DP
chrY    11534648        .       A       G       3.37053e+06     PASS    AC=1284;AF=1;AN=1284;BaseQRankSum=0;DP=134565;FS=30.485;MLEAC=1599;MLEAF=1;MQ=38.34;MQ0=0;MQRankSum=-5.675;NEGATIVE_TRAIN_SITE=����&�:m;QD=25.55;ReadPosRankSum=-2.103;SOR=2.682;VQSLOD=-74.48;culprit=DP
chrY    11537107        .       A       G       3.78562e+06     PASS    AC=1285;AF=1;AN=1285;BaseQRankSum=-0.306;DP=144282;FS=86.932;MLEAC=1599;MLEAF=1;MQ=39.08;MQ0=0;MQRankSum=-4.329;NEGATIVE_TRAIN_SITE=+���&�:m;QD=27.43;ReadPosRankSum=-0.576;SOR=5.518;VQSLOD=-93.36;culprit=DP
> grep -A 2 -B 2 'QD=25.4;ReadPosRankSum=0.756;SOR=1.528;VQSLOD=-22.34;culprit=DP' file_2.vcf
chrY    26658321        .       C       T       53445.4 PASS    AC=1;AF=0.003226;AN=247;DP=1084;FS=0;MLEAC=5;MLEAF=0.016;MQ=48.61;MQ0=0;QD=30.96;SOR=0.89;VQSLOD=-2.736;culprit=DP
chrY    26659649        .       C       T       3.20699e+06     PASS    AC=1284;AF=1;AN=1284;BaseQRankSum=0;DP=75765;FS=0;MLEAC=1599;MLEAF=1;MQ=37.27;MQ0=0;MQRankSum=0.541;NEGATIVE_TRAIN_SITE=�~
?o�<m;QD=25.4;ReadPosRankSum=0.756;SOR=1.528;VQSLOD=-22.34;culprit=DP
chrY    26669015        .       C       T       3.95937e+06     PASS    AC=1285;AF=1;AN=1285;BaseQRankSum=0;DP=113875;FS=0;MLEAC=1599;MLEAF=1;MQ=32.36;MQ0=0;MQRankSum=0.091;NEGATIVE_TRAIN_SITE=5^�=o�<m;QD=29.87;ReadPosRankSum=-0.77;SOR=5.887;VQSLOD=-70.24;culprit=DP
chrY    26671988        .       C       T       1.861e+06       PASS    AC=1285;AF=1;AN=1285;BaseQRankSum=0;DP=46991;FS=0;MLEAC=1599;MLEAF=1;MQ=35.58;MQ0=0;MQRankSum=1.16;NEGATIVE_TRAIN_SITE=�z�?o�<m;QD=29.92;ReadPosRankSum=0.409;SOR=4.72;VQSLOD=-11.98;culprit=DP
```

Note that line breaks have been inserted, affecting downstream analysis, and that garbage characters have also been introduced. The line break was definitely inserted with the `norm` call, because if I `grep` for this line in `file_1`, it looks like this:
```
> grep 'QD=25.4;ReadPosRankSum=0.756;SOR=1.528;VQSLOD=-22.34;culprit=DP' file_1.vcf
chrY    26659649        .       C       T       3.20699e+06     PASS    AC=1284;AF=1;AN=1284;BaseQRankSum=0;DP=75765;FS=0;MLEAC=1599;MLEAF=1;MQ=37.27;MQ0=0;MQRankSum=0.541;NEGATIVE_TRAIN_SITE;QD=25.4;ReadPosRankSum=0.756;SOR=1.528;VQSLOD=-22.34;culprit=DP
```

**Comment by slzarate** [comment]
I have recreated this in very small files to help with debugging (attached; file 1 and file 2 correspond to the descriptions above). Note that the original file has genotypes for 3202 samples; these genotypes have been removed for ease of use.

[tiny_vcfs.zip](https://github.com/samtools/bcftools/files/6593843/tiny_vcfs.zip)

**Comment by slzarate** [comment]
I am updating this issue to reflect what we believe to be the root cause of this issue: that `bcftools norm` doesn't handle the `NEGATIVE_TRAIN_SITE` and `POSITIVE_TRAIN_SITE` fields properly (the only INFO fields of this file of type Flag).

**Comment by pd3** [comment]
There were some fixes after 1.12 came out. Can you please try with the latest github version? I was not able to reproduce the problem with it.

**Comment by slzarate** [comment]
Yes, with the latest github version this issue seems to have been resolved. Thanks :)
